pkg/operation/botanist/migration_test.go

```diff
@@ -127,6 +127,22 @@ var _ = Describe("migration", func() {
 			Expect(botanist.WaitUntilExtensionResourcesMigrated(ctx)).To(Succeed())
 		})
 
+		It("should call the Migrate() func of all the required extension components for workerless Shoot", func() {
+			botanist.Shoot.Workerless = true
+			controlPlaneExposure.EXPECT().WaitMigrate(ctx)
+
+			Expect(botanist.WaitUntilExtensionResourcesMigrated(ctx)).To(Succeed())
+		})
+
+		It("should return an error if not all the WaitMigrate() func of all the required extension components succeed (workerless shoot)", func() {
+			botanist.Shoot.Workerless = true
+			controlPlaneExposure.EXPECT().WaitMigrate(ctx).Return(fakeErr)
+
+			err := botanist.WaitUntilExtensionResourcesMigrated(ctx)
+			Expect(err).To(BeAssignableToTypeOf(&multierror.Error{}))
+			Expect(err.(*multierror.Error).Errors).To(ConsistOf(Equal(fakeErr)))
+		})
+
```

Comment:
Adapt this, we skip controlplane exposure too now.
Ref https://github.com/gardener/gardener/blob/a2931f43dbd53ba192b5efe15ddcc6674157a164/pkg/operation/shoot/shoot.go#L341